api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -331,51 +316,11 @@ && hasPrivilege(userContext, candidate.getRequiredPrivilege())) {
 	public boolean checkRequireExpression(Requireable candidate, AppContextModel contextModel) {
 		try {
 			String requireExpression = candidate.getRequire();
-			if (StringUtils.isBlank(requireExpression)) {
```

Comment:
Did you have to change these?

Reply:
These changes better suited GraalVM. I got some test failures for the old code.

Reply:
I am scared of these very many changes. What happens if we just add the dependency and do not do any other change?

Reply:
Without these changes, I am getting test failures by just adding GraalVM. GraalVM and Nashorn handle datatypes like maps differently. We could also change this method like this:

```java
	@Override
	public boolean checkRequireExpression(Requireable candidate, AppContextModel contextModel) {
		try {
			String requireExpression = candidate.getRequire();
			for (Map.Entry<String, Object> e : contextModel.entrySet()) {
				String jsonValue = new ObjectMapper().writeValueAsString(e.getValue());
				javascriptEngine.eval(e.getKey() + " = " + jsonValue);
			}
			return javascriptEngine.eval("(" + requireExpression + ") == true").equals(Boolean.TRUE);
		}
		catch (Exception e) {
			log.error("Failed to evaluate 'require' check for extension " + candidate.getId(), e);
			return false;
		}
	}

```

Reply:
FWIW, every time i review your code, before commenting, i test them with a fully running instance of the 2.x reference application. This very change has led to all the home page dashboard buttons not being loaded.

Reply:
@dkayiwa I updated the pull request. I am getting some test failures. I tried few different ways to get them fixed but none of the seem to work. Can you take a look?

mvn clean  install log: https://pastebin.com/CnzP8AfA

Reply:
I haven't been following this too closely, but agree with @dkayiwa  based on his comments!

Reply:
> I am getting some test failures

Do they still fail even when you compile on Java 8?

Reply:
Yes

Reply:
Same goes for other dependencies apart from those of graal.

Reply:
Do you want to make their scope test??

Reply:
Don't you think so?

Reply:
Changing the scope of  Jakarta.xml and jaxb-runtime to test causes errors like this:

```
[ERROR] testGetAll(org.openmrs.module.appframework.rest.AdministrativeNotificationResourceTest)  Time elapsed: 1.71 s  <<< ERROR!
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'allAppTemplates': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private javax.validation.Validator org.openmrs.module.appframework.repository.AllAppTemplates.validator; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'validatorFactory' defined in URL [file:/home/wikum/Documents/GSOC/OpenMRS/GSOC-2023/openmrs-module-appframework/omod/target/classes/moduleApplicationContext.xml]: Invocation of init method failed; nested exception is java.lang.NoClassDefFoundError: javax/xml/bind/JAXBException
Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: private javax.validation.Validator org.openmrs.module.appframework.repository.AllAppTemplates.validator; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'validatorFactory' defined in URL [file:/home/wikum/Documents/GSOC/OpenMRS/GSOC-2023/openmrs-module-appframework/omod/target/classes/moduleApplicationContext.xml]: Invocation of init method failed; nested exception is java.lang.NoClassDefFoundError: javax/xml/bind/JAXBException
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'validatorFactory' defined in URL [file:/home/wikum/Documents/GSOC/OpenMRS/GSOC-2023/openmrs-module-appframework/omod/target/classes/moduleApplicationContext.xml]: Invocation of init method failed; nested exception is java.lang.NoClassDefFoundError: javax/xml/bind/JAXBException
Caused by: java.lang.NoClassDefFoundError: javax/xml/bind/JAXBException
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.JAXBException
```

Reply:
Errors at compile time?

Reply:
I fixed these errors by adding the dependencies to the appframwork-omod pom